### PR TITLE
Closes #3: Refactor src/common to avoid duplication, generalize models and facilitate evolutions

### LINKED ISSUE #3: Refactor src/common to avoid duplication, generalize models and facilitate evolutions
**Context**

The wp-rocket-e2e project has a great potential to speed up our development processes: as suggested by @CrochetFeve0251 in https://github.com/wp-media/wp-rocket-e2e/issues/1, every developer could leverage it to automate testing and validation of acceptance criteria. This means that many people will perform small updates regularly on this repository. However, it is currently not up to standards, making it difficult to maintain properly overtime.

Several steps must be taken to make this codebase "production ready".

**Issue**

The src/common folder contains methods that can then be used to facilitate test implementation. However, it is currently quite complex and hard to maintain. 

- [ ]  A refacto of this section is needed so that multiple users (developers, QA) can easily make it evolve as the product and testing needs evolve.
- [ ]  The linter from #2 must return no errors when linting src/common. This check must be added in the CI.

**Observations**

Here are some observations that must be investigated and potentially enhanced:
- In test (*.spec.ts), we instantiate new pageUtils(page) and also things like new fileOptimization( page ). But isn't the second one already instantiated in the first one? Could we avoid a duplication here?
- Unify calls to locators: we find this.page.locator(xxx) but also this.locators.xxx which is the same thing: it's hard to follow!
- All src/common/sections looks like it could be abstracted: define a class with a constructor that takes a JSON listing items (similar to how this.selectors is initiated currently). Abstract methods like visit(), toggle( option, checked),  toggleAll, areAllOptionsEnabled() could be defined, without having to re-write them everytime we add an option. Let's think about how the code will be used: someone will add (or remove) an option in the plugin, and do the same in the test software: it needs to be straightforward like adding a few line in a JSON. It seems there is a refinement for the toggle function, depending if there is an "activate" property defined? -> I suggest to write a quick documentation to explain what the structure should be so that we validate it before developing it.

### PR DESCRIPTION
**Pull request by jeawhanlee**
Closes #3 : Refactor the common modules and singularize selectors.

### CODE REVIEW

File: src/common/selectors.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+                checkbox: "#cache_logged_user",
+                target: "label[for=cache_logged_user]"
+            }
+        }
+    },
+    fileOptimization: {
+        parent: "file_optimization",
+        elements: {
+            minifyCss: {
+                checkbox: "#minify_css",
```
Comment (CrochetFeve0251) [inline src/common/selectors.ts]: I am wondering what you gonna execute if a user create that invalid state: 
```json 
{
   checkbox: "#do_caching_mobile_files",
   textbox: "#cache_reject_cookies"
}
```

It wouldn't be easier to add a field `type` that would define which logic we gonna execute?

File: src/common/selectors.ts [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+                target: "label[for=minify_css]",
+                after: async (page: Page, state: boolean): Promise<void> => {
```
Comment (CrochetFeve0251) [inline src/common/selectors.ts:43]: @jeawhanlee I think we miss understood for the the side case where we can use `after` filed was only RUCSS.

For the other we can create a custom filed type that we handle.

If it is not possible we could handle it using an high order function: 
```js
function clickPopUp(string $text) {
   return async (page: Page, state: boolean): Promise<void> => {
                    if (!state) {
                        return;
                    }

                    await page.waitForSelector(`text=${text}`);
                    await page.locator(`text=${text}`).click();
                };
}
```

This would prevent repeating the code multiple times. :D

File: utils/helpers.ts [hunk 3]
```diff
@@ -2,52 +2,34 @@ import os from 'os';
...
 
 switch(os.platform()) { 
     case 'linux': { 
-       install_path = 'wp-env';
+       installPath = 'wp-env';
        break; 
     } 
     default: { 
-       install_path = '.wp-env'; 
+       installPath = '.wp-env';
```
Comment (CrochetFeve0251) [inline utils/helpers.ts:19]: @jeawhanlee  This logic looks linked to `wordpress\env`.
As we don't use it anymore is it still useful?

**Review by CrochetFeve0251 — CHANGES_REQUESTED** [review]

File: utils/helpers.ts (hunk 3 above)
Comment (jeawhanlee) [reply utils/helpers.ts:19]: @CrochetFeve0251  You are right, but there are still some old tests cases that uses related functions which will need refactoring when we start proper cucumber integration, reason I left some of them there for now.

File: src/common/selectors.ts (hunk 2 above)
Comment (jeawhanlee) [reply src/common/selectors.ts:43]: @CrochetFeve0251 I noticed the WET code there too 😁. A high order function would be better.

**Review by CrochetFeve0251 — APPROVED** [review]
